app/models/concerns/thredded/topic_common.rb

```diff
@@ -47,24 +47,45 @@ def unread(user)
           .merge(reads_class.where(reads[:id].eq(nil).or(reads[:read_at].lt(topics[:updated_at]))))
       end
 
+      private
+
+      # @param user [Thredded.user_class]
+      # @return [ByPostableLookup]
+      def read_states_by_topics_lookup(user, null_value)
+        read_states = reflect_on_association(:user_read_states).klass
+          .where(user_id: user.id, postable_id: current_scope.map(&:id))
+        Thredded::TopicCommon::Lookup.new(read_states, null_value)
```

Comment:
@glebm I see what you using the default value for the hash and the Hash.new{} version and preferable to creating the extra abstraction of the class if we don't need it.

However, I don't get what you mean by "The .postable assignment can happen within the inject". The multiple queries we are trying to avoid are the reverse lookup of topic from read_state (used I assume within the views). I haven't checked this, just assuming that's true as I saw them in the views. And perhaps right now not needed for the Follows, but easier to have them in parallel (and will probably end up using this lookup eventually).